vm/hv1/hv1_emulator/src/hv.rs

```diff
@@ -61,20 +60,19 @@ struct MutableHvState {
 }
 
 impl MutableHvState {
-    fn new(protector: Option<Box<dyn VtlProtectHypercallOverlay>>) -> Self {
+    fn new() -> Self {
         Self {
             hypercall: hvdef::hypercall::MsrHypercallContents::new(),
-            hypercall_protector: protector,
 
             guest_os_id: hvdef::hypercall::HvGuestOsId::new(),
             reference_tsc: hvdef::HvRegisterReferenceTsc::new(),
             tsc_sequence: 0,
         }
     }
 
-    fn reset(&mut self) {
-        if let Some(p) = self.hypercall_protector.as_mut() {
-            p.disable_overlay();
+    fn reset(&mut self, overlay_access: &mut Option<&mut dyn VtlProtectHypercallOverlay>) {
```

Comment:
I'd suggest not taking `Option` here and requiring `virt_whp` to provide a no-op impl.